Anoma/Program.lean

```diff
@@ -14,53 +15,53 @@ inductive StorageValue where
   | bytes (b : ByteArray)
 
-inductive Program.ResourceQuery where
-  | queryByObjectId (uid : ObjectId)
+structure Program.ResourceQuery where
+  uid : ObjectId
 
-inductive Program.Error where
-  | networkError (msg : String)
-  | engineError (msg : String)
-  | decryptionError (msg : String)
-  | commitmentError (msg : String)
+inductive Program.Error : Type where
   | identityError (msg : String)
   | storageError (msg : String)
+  | logicFailed
+  | missingLogic (ref : LogicRef)
+  | balanceCheck (msg : String)
   | typeError (msg : String)
   | userError
+  deriving Repr
 
 /-- Represents an Anoma program, as per
   https://forum.anoma.net/t/reifying-the-local-domain-solver-and-controller-in-the-avm
   -/
-inductive Program where
+inductive Program : Type 2 where
   | skip
   | raise (err : Program.Error)
   | tryCatch (prog : Program) (onError : Program.Error → Program) (next : Program)
-  | withRandomGen (next : StdGen → Program × StdGen)
+  | withRandomGen (next : StdGen → Program)
   | queryResource (query : Program.ResourceQuery) (next : Resource → Program)
   | submitTransaction (tx : Transaction) (next : Program)
-  | decrypt (engine : EngineId) (data : Ciphertext) (next : Plaintext → Program)
-  | requestCommitment (engine : EngineId) (data : Signable) (next : Commitment → Program)
-  | generateIdentity (backend : Backend) (params : IDParams) (capabilities : Capabilities)
-      (next : (commitmentEngine : Option EngineId) → (decryptionEngine : Option EngineId) → (externalIdentity : EngineId) → Program)
-  | connectIdentity (externalIdentity : EngineId) (backend : Backend) (capabilities : Capabilities)
-      (next : (commitmentEngine : Option EngineId) → (decryptionEngine : Option EngineId) → Program)
-  | deleteIdentity (externalIdentity : EngineId) (backend : Backend) (next : Program)
-  | getValue (key : StorageKey) (next : Option StorageValue → Program)
-  | setValue (key : StorageKey) (value : StorageValue) (next : Program)
-  | deleteValue (key : StorageKey) (next : Program)
+  | log (msg : String) (next : Program)
+  -- | decrypt (engine : EngineId) (data : Ciphertext) (next : Plaintext → Program)
+  -- | requestCommitment (engine : EngineId) (data : Signable) (next : Commitment → Program)
+  -- | generateIdentity (backend : Backend) (params : IDParams) (capabilities : Capabilities)
+  --     (next : (commitmentEngine : Option EngineId) → (decryptionEngine : Option EngineId) → (externalIdentity : EngineId) → Program)
+  -- | connectIdentity (externalIdentity : EngineId) (backend : Backend) (capabilities : Capabilities)
+  --     (next : (commitmentEngine : Option EngineId) → (decryptionEngine : Option EngineId) → Program)
+  -- | deleteIdentity (externalIdentity : EngineId) (backend : Backend) (next : Program)
+  -- | getValue (key : StorageKey) (next : Option StorageValue → Program)
+  -- | setValue (key : StorageKey) (value : StorageValue) (next : Program)
+  -- | deleteValue (key : StorageKey) (next : Program)
```

Comment:
Maybe let's better just remove those? We can always add later, and there's some chance the specs may change in the meantime anyway.

Reply:
I've removed them